From 838c9c6efb376028d6b09a170a89373276bc45d4 Mon Sep 17 00:00:00 2001
From: "George G. Davis" <george_davis@mentor.com>
Date: Mon, 8 Jan 2018 13:38:25 -0500
Subject: [PATCH 162/909] soc: renesas: rcar_ems_ctrl: Fix "Using plain integer
 as NULL pointer"

commit 4f1b5cb87ae6b4751e3e246ab0424d8049290720 from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

The following sparse (`make C=1`) issues are reported:

drivers/soc/renesas/rcar_ems_ctrl.c:259:64: warning: Using plain integer as NULL pointer
drivers/soc/renesas/rcar_ems_ctrl.c:296:44: warning: Using plain integer as NULL pointer

Fix the above sparse issues by replacing `0` with `NULL`.

Signed-off-by: George G. Davis <george_davis@mentor.com>
Signed-off-by: Steve Longerbeam <steve_longerbeam@mentor.com>
Signed-off-by: Dien Pham <dien.pham.ry@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/soc/renesas/rcar_ems_ctrl.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/soc/renesas/rcar_ems_ctrl.c b/drivers/soc/renesas/rcar_ems_ctrl.c
index 8fda8f92c9db..7aa976002e44 100644
--- a/drivers/soc/renesas/rcar_ems_ctrl.c
+++ b/drivers/soc/renesas/rcar_ems_ctrl.c
@@ -256,7 +256,8 @@ static int rcar_ems_cpu_shutdown_init(void)
 		return 0;
 
 	total_target_cpu = of_count_phandle_with_args(ems_node,
-						"target_cpus", 0);
+						"target_cpus", NULL);
+
 	for_each_online_cpu(cpu) {
 		tmp_node  = of_get_cpu_node(cpu, NULL);
 		for (i = 0; i < total_target_cpu; i++) {
@@ -290,7 +291,7 @@ static int rcar_ems_cpu_shutdown_init(void)
 
 static void rcar_ems_cpu_shutdown_exit(void)
 {
-	rcar_ems_notify(RCAR_EMS_MODE_OFF, 0);
+	rcar_ems_notify(RCAR_EMS_MODE_OFF, NULL);
 	unregister_rcar_ems_notifier(&ems_thermal_notifier_block);
 	cpufreq_unregister_notifier(&ems_cpufreq_notifier_block,
 				    CPUFREQ_POLICY_NOTIFIER);
-- 
2.17.1

